pkg/registrar/registrar.go

```diff
@@ -146,6 +152,10 @@ func (r *Registrar) register(ctx context.Context, cl zbus.Client, env environmen
 			if err := r.reActivate(ctx, cl, env); err != nil {
 				log.Error().Err(err).Msg("failed to reactivate account")
 			}
+		case <-time.After(updateLocationInterval):
+			if err := r.updateLocation(ctx, cl); err != nil {
```

Comment:
Code in general looks way better. But why just not call `register()` here. It already collects the location, then compare if the node information has changed before calling update.